components/molecule/dropdownList/src/index.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+const BASE_CLASS = `sui-MoleculeDropdownList`
+const CLASS_HIDDEN = `${BASE_CLASS}--hidden`
+
+const SIZES = {
+  SMALL: 'small',
+  MEDIUM: 'medium',
+  LARGE: 'large'
+}
+
+class MoleculeDropdownList extends Component {
+  refDropdownList = React.createRef()
+
+  get extendedChildren() {
+    const {children, value, size, visible, onChange, ...props} = this.props
+    return React.Children.toArray(children)
+      .filter(Boolean)
+      .map((child, index) => {
+        const {value: valueChild} = child.props
+        const selected = Array.isArray(value)
+          ? value.includes(valueChild)
+          : value === valueChild
+        return React.cloneElement(child, {
+          ...props,
+          index,
+          onClick: onChange,
+          selected
+        })
+      })
+  }
+
+  get classNames() {
+    const {size, visible} = this.props
+    return cx(BASE_CLASS, `${BASE_CLASS}--${size}`, {
+      [CLASS_HIDDEN]: !visible
+    })
+  }
+
+  handleKeyDown = e => {
+    const items = this.refDropdownList.current.children
+    const {key} = e
+    if (key === 'ArrowDown' || key === 'ArrowUp') {
+      const currentElementFocused = document.activeElement
+      const index = Array.from(items).reduce((acc, option, index) => {
+        if (option === currentElementFocused) acc = index
+        return acc
+      }, 0)
+      if (index >= 0 || index <= items.length) {
```

Comment:
It's not exactly the same. Mind the difference between `index <= numOptions` in one condition and `index < numOptions` in the other one

Taking into account it enters w/ this condition `index <= numOptions`
- I don't want to `options[index + 1].focus()` if `index === numOptions`. Only if `index < numOptions`
- But it may want to `options[index - 1].focus()` if `index === numOptions`